eth/catalyst/api_test.go

```diff
@@ -1233,26 +1238,53 @@ func TestNilWithdrawals(t *testing.T) {
 }
 
 func setupBodies(t *testing.T) (*node.Node, *eth.Ethereum, []*types.Block) {
-	genesis, preMergeBlocks := generateMergeChain(10, false)
+	genesis, preMergeBlocks := generateMergeChain(10, true)
 	n, ethservice := startEthService(t, genesis, preMergeBlocks)
+	// enable shanghai on the last block
+	ethservice.BlockChain().Config().ShanghaiTime = &preMergeBlocks[len(preMergeBlocks)-1].Header().Time
```

Comment:
IF shanghai time is `==` that header time, doesn't that mean that _that_ particular header is also shanghai?